src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -994,4 +994,21 @@ final case class AgreementApiServiceImpl(
       removeAgreementConsumerDocumentResponse[Unit](operationLabel)(_ => removeAgreementConsumerDocument204)
     }
   }
+
+  override def getAgreementProducers(q: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCompactOrganizations: ToEntityMarshaller[CompactOrganizations]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE) {
```

Comment:
```suggestion
  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE) {
```

Reply:
a new test is required to verify roles for this endpoint.
You can find these tests under `src/test/scala/it/pagopa/interop/agreementprocess/AgreementApiAuthzSpec.scala`